jenkins/Jenkinsfile

```diff
@@ -0,0 +1,256 @@
+def install_test_dependencies(String cache_prefix) {
+  /* restore cache with key: << parameters.cache_prefix >>-dependencies-{{ checksum "dev-requirements.txt" }}-{{ checksum "test-requirements.txt" }}-{{ checksum "setup.py" }} */
+  sh '''#!/bin/bash
+  pip install -r dev-requirements.txt --user
+  pip install -r test-requirements.txt --user
+  pip install -e '.[dispatcher]' --user
+  '''
+  /* save cache with relevant paths and key */
+}
+def pytest(String target){
+  echo "running pytest << ${target} >>"
+  sh """#!/bin/bash
+  pytest \
+    -n 4 \
+    --cov-report term-missing \
+    --cov= ${target} \
+    ${target}/tests \
+    --junitxml=test-results/ ${target}
+  """
+}
+def build_docs(String project){
+  echo "build docs for ${project}"
+  sh """
+  #!/bin/bash
+  sphinx-build \
+    -d _build/doctrees \
+    -D version="\$(python setup.py --version)" \
+    docs/${project} \
+    docs/${project}/_build/html
+  """
+}
+@Library('pipeline-shared-library') _
+pipeline {
+  parameters {
+    string(name: 'VERSION', defaultValue: '5.1.0', description: 'Cloudify version label, to be used in the filenames (eg. "5.1.0")')
+    string(name: 'PRERELEASE', defaultValue: '.dev1', description: 'A prerelase/milestone label, to be used in the filenames (eg. ".dev1")')
+    string(name: 'BRANCH', defaultValue: 'master', description: 'Branch name for cloudify-agent')
+  }
+  agent {
+    kubernetes {
+      label 'common'
+      defaultContainer 'jnlp'
+      yamlFile 'jenkins/build-pod.yaml'
+    }
+  }
+  options {
+    checkoutToSubdirectory('cloudify-common')
+    buildDiscarder(logRotator(numToKeepStr:'30'))
+    timeout(time: 60, unit: 'MINUTES')
+    timestamps()
+  }
+  environment {
+    VERSION = "${params.VERSION}"
+    PRERELEASE = "${params.PRERELEASE}"
+    BRANCH = "${params.BRANCH}"
+    UPLOAD_S3_BUILD_NUMBER = "${params.UPSTREAM_BUILD_NUMBER}"
+    WORKSPACE = "${env.WORKSPACE}"
+    PROJECT = "cloudify-common"
+    UPLOAD_FOLDER = 'cloudify-common'
+    STAGE_DIR = "cloudify-common"
+    PATH= "/root/.local/bin:$PATH"
+    CFY_MANAGER_URL = 'https://raw.githubusercontent.com/cloudify-cosmo/cloudify-manager'
+    CFY_NODE_RPM = 'http://repository.cloudifysource.org/cloudify/components/nodejs-12.16.1-1nodesource.x86_64.rpm'
+  }
+  stages{
```

Comment:
Go over each stage in Jenkins and validate that the output is pretty,
you can see in the link below it's all `Print message` other jobs are better but still can be improved

https://jenkins.eks.cloudify.co/blue/organizations/jenkins/Common-Multibranch/detail/yuval-build/3/pipeline/192